shell/platform/android/test/io/flutter/util/PreconditionsTest.java

```diff
@@ -0,0 +1,34 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.util;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertThrows;
+
+import android.text.TextUtils;
+import io.flutter.util.Preconditions;
+import org.junit.runner.RunWith;
+import org.junit.Test;
+import org.robolectric.annotation.Config;
+import org.robolectric.RobolectricTestRunner;
+
+@Config(manifest=Config.NONE)
+@RunWith(RobolectricTestRunner.class)
+public class PreconditionsTest {
+    @Test
+    public void checkNotNull_notNull() {
+        // Should always return its input.
+        assertEquals("non-null", Preconditions.checkNotNull("non-null"));
+        assertEquals(42, (int) Preconditions.checkNotNull(42));
+        Object classParam = new Object();
+        assertEquals(classParam, Preconditions.checkNotNull(classParam));
+    }
+
+    @Test
+    public void checkNotNull_Null() {
+        assertThrows(NullPointerException.class, () -> {Preconditions.checkNotNull(null);});
+    }
+}
```

Comment:
Ditto.

For this file, I'm not sure what more it's really adding beyond the smoke test - do we need it?

Reply:
I don't think the test is critical, but it's proving a couple things:

1. We can compile the tests with Java 8 like everything else (lambda in the `assertThrows`).
2. We're importing an io.flutter class into a test and running it on that.
3. Having two classes at once in this PR shows what the TestSuite is actually for and proves that it's working and running multiple classes.

Even though it's not critical this is also actually a working test that would keep this function from regressing, however unlikely. I don't think it's necessary but I don't think it's bad to have either. WDYT?